packages/tradershub/src/features/cfd/modals/MT5PasswordModal/MT5PasswordModal.tsx

```diff
@@ -0,0 +1,304 @@
+    useActiveTradingAccount,
+    useAvailableMT5Accounts,
+    useCreateMT5Account,
+    useMT5AccountsList,
+    useSettings,
+    useTradingPlatformPasswordChange,
+} from '@deriv/api';
+import { Provider } from '@deriv/library';
+import { Button, useBreakpoint } from '@deriv/quill-design';
+import { ActionScreen, ButtonGroup, Dialog, ModalStepWrapper } from '../../../../components';
+import MT5PasswordIcon from '../../../../public/images/ic-mt5-password.svg';
+import { TMarketTypes, TPlatforms } from '../../../../types';
+import { validPassword } from '../../../../utils/password';
+import { companyNamesAndUrls, MarketTypeDetails, PlatformDetails } from '../../constants';
+import { CFDSuccess, CreatePassword, EnterPassword } from '../../screens';
+
+type TProps = {
+    marketType: TMarketTypes.SortedMT5Accounts;
+    platform: TPlatforms.All;
+};
+
+const MT5PasswordModal: FC<TProps> = ({ marketType, platform }) => {
+    const [password, setPassword] = useState('');
+    const { error, isLoading: createMT5AccountLoading, isSuccess, mutate, status } = useCreateMT5Account();
+    const { isLoading: tradingPlatformPasswordChangeLoading, mutate: tradingPasswordChange } =
+        useTradingPlatformPasswordChange();
+    const { data: accountStatus } = useAccountStatus();
+    const { data: ActiveTradingAccount } = useActiveTradingAccount();
+    const { data: mt5Accounts } = useMT5AccountsList();
+    const { data: availableMT5Accounts } = useAvailableMT5Accounts();
+    const { data: settings } = useSettings();
+    const { getModalState, hide, show } = Provider.useModal();
+    const { isMobile } = useBreakpoint();
+    const history = useHistory();
+
+    const isMT5PasswordNotSet = accountStatus?.is_mt5_password_not_set;
+
+    const hasMT5Account = mt5Accounts?.find(account => account.login);
+    const isDemo = ActiveTradingAccount?.is_virtual;
+    const marketTypeTitle =
+        marketType === 'all' && Object.keys(PlatformDetails).includes(platform)
+            ? PlatformDetails[platform].title
+            : MarketTypeDetails[marketType].title;
+    const selectedJurisdiction = getModalState('selectedJurisdiction');
+
+    const landingCompanyName = `(${
+        companyNamesAndUrls?.[selectedJurisdiction as keyof typeof companyNamesAndUrls]?.shortcode
+    })`;
+
+    const onSubmit = useCallback(async () => {
+        const accountType = marketType === 'synthetic' ? 'gaming' : marketType;
+
+        // in order to create account, we need to set a password through trading_platform_password_change endpoint first
+        // then only mt5_create_account can be called, otherwise it will response an error for password required
+        if (isMT5PasswordNotSet) {
+            await tradingPasswordChange({
+                new_password: password,
+                platform: PlatformDetails.mt5.platform,
+            });
+        }
+
+        const categoryAccountType = ActiveTradingAccount?.is_virtual ? 'demo' : accountType;
```

Comment:
rename it to pascalCase
```suggestion
        const categoryAccountType = activeTradingAccount?.is_virtual ? 'demo' : accountType;
```